applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1068,6 +1132,10 @@ static void disconnect(const void *subscriber_id, uint8_t report_id)
 			report_send(rd, true, true);
 		}
 	}
+
+	if (update_kbd_leds) {
+		broadcast_keyboard_leds(get_keyboard_subscriber());
```

Comment:
you could call `` regardless as the function has internal `update_needed`. This call will do nothing if update is not needed.

Reply:
You are right. Changed.

Reply:
I simplified code a bit. The HID output reports related to keyboard LEDs are not handled too often.